glusterd2/volume/volume-utils.go

```diff
@@ -191,6 +191,10 @@ func CreateVolumeInfoResp(v *Volinfo) *api.VolumeInfo {
 		Options:   v.Options,
 		Subvols:   CreateSubvolInfo(&v.Subvols),
 		Metadata:  v.Metadata,
-		SnapList:  v.SnapList,
+		SnapList:  make([]string, len(v.SnapList)),
 	}
+	for i, snap := range v.SnapList {
```

Comment:
Instead of iterating, how about checking the `null` itself?

    if volinfo.SnapList == nil {
        volinfo.SnapList = []string{}
    }

Reply:
yeah, This will be simpler.